test/controllers/agent/TaxYearsControllerSpec.scala

```diff
@@ -92,19 +92,6 @@ class TaxYearsControllerSpec extends TestSupport
 				result.getMessage shouldBe "[ClientConfirmedController][getMtdItUserWithIncomeSources] IncomeSourceDetailsModel not created"
 			}
 		}
-		"there is no firstAccountingPeriodEndDate from income source details" should {
```

Comment:
This should be moved to the IncomeSourceDetailsResponse.scala  spec, along with some new tests for the different scenarios possible.